src/Counter.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+class Counter extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = { count: 5 };
+
+        this.changeCounter = this.changeCounter.bind(this);
+    }
+
+    changeCounter(increase) {
+        let {count} = this.state;
+        count = increase ? count + 1 : count - 1;
+        if (count < 0) count = 0;
+
+        this.setState({ count });
```

Comment:
It's not an error, but I've recommend to not update state when `count` is less then 0, because after each state updating, component will rerendered